salesforce_api/gsheet-new-row-to-sfdc-new-contact/sample.bal

```diff
@@ -0,0 +1,52 @@
+import ballerina/log;
+import ballerinax/googleapis.sheets;
+import ballerinax/salesforce as sfdc;
+
+public type Contact record {|
+    string Id;
+    string Email;
+|};
+
+const int HEADINGS_ROW = 1;
+
+// Google sheets configuration parameters
+configurable string spreadsheetId = ?;
+configurable string worksheetName = ?;
+configurable string duplicateWorksheetName = ?;
+configurable string gSheetsAccessToken = ?;
+
+// Salesforce configuration parameters
+configurable string salesforceAccessToken = ?;
+configurable string salesforceBaseUrl = ?;
+
+sheets:Client sheetsClient = check new ({auth: {token: gSheetsAccessToken}});
+
+sfdc:Client sfdcClient = check new ({
+    baseUrl: salesforceBaseUrl,
+    auth: {
+        token: salesforceAccessToken
+    }
+});
+
+public function main() returns error? {
+    sheets:Range range = check sheetsClient->getRange(spreadsheetId, worksheetName, "A1:G");
+    (int|string|decimal)[] headers = range.values[0];
+    foreach (int|string|decimal)[] item in range.values.slice(HEADINGS_ROW) {
+        int? indexOfEmail = headers.indexOf("Email");
+        if indexOfEmail is () {
+            return error("Email column not found");
+        }
+        stream<Contact, error?> retrievedStream = check sfdcClient->query(string `SELECT Id, Email FROM Contact WHERE Email='${item[indexOfEmail]}'`);
```

Comment:
Isn't it better to extract `item[indexOfEmail` to a local variable and use it in three usages.

Reply:
Yes it's much more efficient but this will need a casting (to string) as the value returned is a `int|string|decimal`